core/templates/dev/head/domain/exploration/AnswerGroupObjectFactory.js

```diff
@@ -18,9 +18,10 @@
  */
 
 oppia.factory('AnswerGroupObjectFactory', [function() {
-  var AnswerGroup = function(ruleSpecs, outcome, correct) {
-    this.ruleSpecs = ruleSpecs;
-    this.outcome = outcome;
+  var AnswerGroup = function(ruleSpecsBackendList, outcomeBackendDict,
+    correct) {
+    this.ruleSpecs = ruleSpecsBackendList;
```

Comment:
Maybe we should use a different word to capture anything that is not a front-end domain object lol